code/a/validate_con_imputation.do

```diff
@@ -0,0 +1,148 @@
+
+4. graph: share shrids dropped on X axis, mean absolute deviation on Y axis.
+
+*/
+
+/**********************************************************************************************************************/
+/* program impute_shrid_pop : Impute population for missing villages based on other villages in the same constituency */
+/*   note: this is copy-pasted from collapse_shrug_to_con.do exactly */
+/**********************************************************************************************************************/
+cap prog drop impute_shrid_pop
+prog def impute_shrid_pop
+{
+  syntax, group_id(string) base_year(string) target_year(string) [pca_suffix(string)]
+
+  /* example: base_year: 11
+            target_year: 01
+             pca_suffix: _r  [can be blank for U + R]
+  */
+  
+  
+  /* calculate total non-missing constituency population in target year */
+  bys `group_id': egen c_pop`target_year'_orig`pca_suffix' = total(pc`target_year'_pca_tot_p`pca_suffix')
+  
+  /* by constituency, calculate base year population in places with non-missing target year population */
+  bys `group_id': egen c_pop`base_year'_found`target_year'`pca_suffix' = total(pc`base_year'_pca_tot_p`pca_suffix' * !mi(pc`target_year'_pca_tot_p`pca_suffix'))
+  
+  /* these two variables give us an estimate of population growth in this constituency */
+  /* we use this to predict target (2001) population from the base (2011) population */
+  /* the scaling factor (relative to 2011) is the con 2001 pop divided by the
+     con 2011 pop * in the places where 2001 was non-missing* */
+  /* same as dividing 2011 pop by the average growth rate from 2001-2011 in this con */
+  replace pc`target_year'_pca_tot_p`pca_suffix' = (pc`base_year'_pca_tot_p`pca_suffix') * c_pop`target_year'_orig`pca_suffix' / c_pop`base_year'_found`target_year'`pca_suffix' if mi(pc`target_year'_pca_tot_p`pca_suffix') & mi(pc`target_year'_pca_tot_p)
+
+  /* note the double missing requirement -- e.g. need to be missing
+  tot_p_r AND tot_p in order to impute rural population -- otherwise
+  this could be a town that became a village or vice versa */
+
+}
+end
+/* *********** END program impute_shrid_pop ***************************************** */
+
+/* open the shrid-con key, which is unique on shrids */
+use $shrug/keys/shrug_con_key_2008, clear
+
+/* limit the sample to places where we observe pc11 population */
+get_shrug_var pc11_pca_tot_p 
+drop if mi(pc11_pca_tot_p)
+
+// /* drop constituencies where total pc11 population is too big or small */
+// get_shrug_key pc11_state_id 
+// merge m:1 pc11_state_id using $tmp/mean_con_pop, keep(match master) keepusing(mean_con_pop)
+// 
+// /* get total constituency population based on shrid match */
+// bys ac08_id: egen con_pop = total(pc11_pca_tot_p)
+// tag ac08_id
+// gen con_pop_ratio = con_pop / mean_con_pop
+// assert !mi(con_pop_ratio)
+// drop if !inrange(con_pop_ratio, .4, 2.5)
+// drop con_pop con_pop_ratio mean_con_pop 
+
```

Comment:
delete

Reply:
fixed in this [commit](https://github.com/devdatalab/paper-shrug/commit/19a059b092ad2fa54f49dccc7f3808daae340eb1)